Public/Src/Cache/ContentStore/Distributed/NuCache/LocalLocationStore.cs

```diff
@@ -1187,67 +1187,95 @@ public async Task<ReconciliationResult> ReconcileAsync(OperationContext context)
 
                     token.ThrowIfCancellationRequested();
 
-                    // Pause events in main event store while sending reconciliation events via temporary event store
-                    // to ensure reconciliation does cause some content to be lost due to apply reconciliation changes
-                    // in the wrong order. For instance, if a machine has content [A] and [A] is removed during reconciliation.
-                    // It is possible that remove event could be sent before reconciliation event and the final state
-                    // in the database would still have missing content [A].
-                    using (EventStore.PauseSendingEvents())
+                    var totalAddedContent = 0;
+                    var totalRemovedContent = 0;
+                    var allLocalStoreContentCount = 0;
+                    ShortHash? lastProcessedHash = null;
+                    var isFinished = false;
+
+                    while (!isFinished)
                     {
-                        var allLocalStoreContentInfos = await _localContentStore.GetContentInfoAsync(token);
-                        token.ThrowIfCancellationRequested();
+                        var delayTask = Task.Delay(_configuration.ReconciliationCycleFrequency);

```

Comment:
>var delayTask = Task.Delay(_configuration.ReconciliationCycleFrequency); [](start = 24, length = 72)

Why create the delay task here?

Reply:
Wanted to make sure that we started the cycle exacly every {frequency}, instead of every {frequency + operation duration} in the case when I create the task after operation completion. I realize that probably the operation duration is fairly quick, but still I took the precaution.

---
In reply to: [335053567](https://github.com/microsoft/BuildXL/pull/1046#discussion_r335053567) [](ancestors = 335053567)